relay-pattern/src/lib.rs

```diff
@@ -731,6 +760,12 @@ enum Token {
     Class { negated: bool, ranges: Ranges },
     /// A list of nested alternate tokens `{a,b}`.
     Alternates(Vec<Tokens>),
+    /// A list of optional tokens.
+    ///
+    /// This has no syntax of its own, it's parsed
+    /// from alternatives containing empty branches
+    /// like `{a,b,}`.
+    Optional(Vec<Token>),
```

Comment:
```suggestion
    Optional(Tokens),
```

Tokens gives additional guarantees, we should make sure the tokens contained here uphold the same guarantees.